src/lib/discord/roles/addRole.ts

```diff
@@ -0,0 +1,26 @@
+// applies a role to a given user
+export async function addRole(
+  roleId: string | undefined,
+  guildId: string | undefined,
+  userId: string
+) {
+  const res = await fetch(
```

Comment:
We should be able to use `api.put(Routes.guildMemberRole(...))` from `$discord/api` and `discord-api-types/v10`